### DISCUSSION TITLE
Less check for updates ?

### DISCUSSION THREAD

**Discussion by LaurentOngaro** [imjp94/gd-plug-ui#3] [Ideas]
I've noticed that a check for plugins updates is done very often (I've added some log prints into the update hook). 
I'm not sure it's useful , and it could slow down the editor if lots of plugins are managed by gd-plug.

On my own, I've disabled this feature by making some hack to your code in the following cases:
- when the visibility of the tab changes (I prefer a manual click on the "check for update" button)
- each time the project is run (I think it's not useful to check for update "outside" of a tool/editor context)

If you want, I can create a fork and submit a PR for that

**Comment by imjp94** [comment]
Do you mean the update of gd-plug-ui editor(`PluginSettings.update_plugin_list`)?
https://github.com/imjp94/gd-plug-ui/blob/b1a97ab05f8940d08795dbaa157448fee42ce2a4/addons/gd-plug-ui/scene/plugin_settings/PluginSettings.gd#L94

Otherwise, gd-plug update hook will only be called when the plugin is updated.

**Comment by LaurentOngaro** [comment]
I've rather disabled (ie commented out) the call to load_gd_plug() in these cases.

As gd_plug._plugging() is called in load_gd_plug(), this could be "heavy"  in these cases because each add on is checked and updated if needed.

**Reply by imjp94** [reply]
I see. `gd_plug._plugging` is actually quite fast, since it only check and validate user defined plugins with `gd_plug.plug`(this method doesn't update anything)

But optimization is welcomed, as the frequent calling of `load_gd_plug()` is caused by PluginSettings.visible always return true even when the Project Setting window is closed.
